cupy/indexing/generate.py

```diff
@@ -1,9 +1,160 @@
+        return newobj
+
+    def __init__(self, axis=0, matrix=False, ndmin=1, trans1d=-1):
+        self._axis = axis
+        self._matrix = matrix
+        self.axis = axis
+        self.matrix = matrix
+        self.col = 0
+        self.trans1d = trans1d
+        self.ndmin = ndmin
+
+    def __getitem__(self, key):
+        trans1d = self.trans1d
+        ndmin = self.ndmin
+        objs = []
+        if isinstance(key, str):
+            return NotImplemented
+        if not isinstance(key, tuple):
+            key = (key,)
+
+        for k in six.moves.range(len(key)):
+            if isinstance(key[k], slice):
+                return NotImplemented
+            elif isinstance(key[k], str):
+                if k != 0:
+                    raise ValueError(
+                    'special directives must be the first entry.')
+                return NotImplemented
+            else:
+                newobj = key[k]
+                if ndmin > 1:
+                    tempobj = cupy.array(newobj, copy=False)
+                    newobj = cupy.array(newobj, copy=False, ndmin=ndmin)
+                    if trans1d != -1 and tempobj.ndim < ndmin:
+                        newobj = self._output_obj(newobj, ndmin, trans1d)
+                    del tempobj
+                elif ndmin == 1:
+                    tempobj = cupy.array(newobj, copy=False)
+                    newobj = cupy.array(newobj, copy=False, ndmin=ndmin)
+                    if tempobj.ndim < ndmin:
+                        newobj = self._output_obj(newobj, tempobj, ndmin, trans1d)
+                    del tempobj
+
+            objs.append(newobj)
+
+        res = cupy.concatenate(tuple(objs), axis=self.axis)
+        return self._retval(res)
+
+    def __len__(self):
+        return 0
+
+class CClass(AxisConcatenator):
+    def __init__(self):
+        """Translates slice objects to concatenation along the second axis.
+
+        This is short-hand for ``np.r_['-1,2,0', index expression]``, which is
+        useful because of its common occurrence. In particular, arrays will be
+        stacked along their last axis after being upgraded to at least 2-D with
+        1's post-pended to the shape (column vectors made out of 1-D arrays).
+
+        For detailed documentation, see `r_`.
+        """
+        AxisConcatenator.__init__(self, -1, ndmin=2, trans1d=0)
+
+c_ = CClass()
+
+
+class RClass(AxisConcatenator):
+    def __init__(self):
+        """Translates slice objects to concatenation along the first axis.
+
+        This is a simple way to build up arrays quickly. There are two use cases.
+
+        1. If the index expression contains comma separated arrays, then stack
+           them along their first axis.
+        2. If the index expression contains slice notation or scalars then create
+           a 1-D array with a range indicated by the slice notation.(Not Implemented)
+
+        If slice notation is used, the syntax ``start:stop:step`` is equivalent
```

Comment:
Features described in this paragraph are not implemented yet. So please remove this paragraph or explicitly write that the features are not implemented.